README.md

```diff
@@ -21,238 +24,131 @@ repositories {
+    // RX support
+    implementation 'com.github.jacek-marchwicki.recyclerview-changes-detector:universal-adapter-rx:<look-on-release-tab>'
 }
 ```
 
 ## How to use
 
-### With Universal Adapter
-
-Implement some Pojo with your data:
+Let's assume that your list consits of a header, songs and footer.
 
-```java
-private class Data implements BaseAdapterItem {
+- Implement data models for all list elements
 
-    private final long id;
-    @Nonnull
-    private final String name;
-    private final int color;
-
-    Data(long id, @Nonnull String name, int color) {
-        this.id = id;
-        this.name = name;
-        this.color = color;
-    }
-
-    @Override
-    public long adapterId() {
-        return id;
-    }
-
-    /**
-     * Return true if id matches
-     */
-    @Override
-    public boolean matches(@Nonnull BaseAdapterItem item) {
-        return item instanceof Data && (((Data) item).id == id);
-    }
-
-    /**
-     * Return true if items are equal
-     */
-    @Override
-    public boolean same(@Nonnull BaseAdapterItem item) {
-        return equals(item);
-    }
+```kotlin
+data class HeaderItem(val text: String, val songsCount: Int, override val itemId: Any = text) : DefaultAdapterItem()
+data class SongItem(val id: String, val title: String, val imageUrl: String, override val itemId: Any = id, val onSongClick: (id: String) -> Unit) : DefaultAdapterItem()
+data class Footer(override val itemId: Any = NO_ID) : DefaultAdapterItem()
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (!(o instanceof Data)) return false;
-        final Data data = (Data) o;
-        return id == data.id &&
-                name.equals(data.name) &&
-                color == data.color;
-    }
-}
 ```
 
-Implement your holder:
-
-```java
-public class DataViewHolder implements ViewHolderManager {
-
-    @Override
-    public boolean matches(@Nonnull BaseAdapterItem baseAdapterItem) {
-        return baseAdapterItem instanceof Data;
-    }
-
-    @Nonnull
-    @Override
-    public BaseViewHolder createViewHolder(@Nonnull ViewGroup parent, @Nonnull LayoutInflater inflater) {
-        return new ViewHolder(inflater.inflate(R.layout.data_item, parent, false));
-    }
-
-    private class ViewHolder extends BaseViewHolder<Data> {
+- Implement a view holder for each data model to bind its data to the view. You have to specify item layout id and data model class.
 
-        @Nonnull
-        private final TextView text;
-        @Nonnull
-        private final CardView cardView;
-
-        ViewHolder(@Nonnull View itemView) {
-            super(itemView);
-            text = (TextView) itemView.findViewById(R.id.data_item_text);
-            cardView = (CardView) itemView.findViewById(R.id.data_item_cardview);
+```kotlin
+class HeaderViewHolder : LayoutViewHolderManager<HeaderItem>(
```

Comment:
suggestion: I think it is worth having exactly the same sample code at:
* the provided GIF
* provided example https://github.com/jacek-marchwicki/recyclerview-changes-detector/blob/master/app/src/main/java/com/jacekmarchwicki/changesdetector/example/MainActivity.kt
* in README

Otherwise reader can be confused

Reply:
Good point, it should be the same at least in the GIF ans READMd